src/index/src/fulltext_index/create/tantivy.rs

```diff
@@ -59,10 +68,17 @@ impl TantivyFulltextIndexCreator {
 
         let memory_limit = Self::sanitize_memory_limit(memory_limit);
 
-        let writer = SingleSegmentIndexWriter::new(index, memory_limit).context(TantivySnafu)?;
+        // Use one thread to keep order of the row id.
+        let writer = index
+            .writer_with_num_threads(1, memory_limit)
```

Comment:
How many threads tantivy uses if we don't set it? Will it affect performance?

Reply:
When using `SingleSegmentIndexWriter`, one thread is used as well, and there is not much change in our tests.